### DISCUSSION TITLE
Looping from Javascript in Automations

### DISCUSSION THREAD

**Discussion by AlexandriaLittle** [Budibase/budibase#9856] [Help]
Hi All,

I'm running into a roadblock with an automation. Here's the layout:
![image](https://user-images.githubusercontent.com/60404243/222302313-8231bc82-b9cd-4e08-8752-f718469b3e54.png)

 
I have a simple Javascript step in the automation that pulls out nested member data from an API's JSON response. Here's the code:

```
const items = {{steps[2].items}};
let results = [];
for (let item of items){
	for (let response of item["response"]) {
		for (let member of response["members"]){
			results.push(member);
		}
	}
}

return results;
```

The issue I'm having is when I try to loop over the data in the next step to create rows the data doesn't get looped over. Here's the Create Row section with looping:
![image](https://user-images.githubusercontent.com/60404243/222302623-db4772e7-1450-48e5-b5f6-677877efc568.png)

I'm not sure what to do from here to get it to work.

**Comment by melohagan** [comment]
Hey @AlexandriaLittle 

This line in the JavaScript looks incorrect to me: `const items = {{steps[2].items}};`

It should just be `const items = steps[2].items;`. If that doesn't work maybe try `steps[3].items` as it appears to be Step 3 from the screenshot

**Reply by AlexandriaLittle** [reply]
I fixed that error but still no luck. The output from the javascript step looks correct - it's outputting an array full of JSON objects. I just get nothing for luck when I loop before step 6.

**Reply by melohagan** [reply]
Should you not be looping over the response from Step 4, i.e. the JavaScript block output?

Your screenshot show `{{ steps.3.value }}` which would be the output of External Data Connector.

**Reply by AlexandriaLittle** [reply]
I gave that a shot and I was given an error in the developer console:
![image](https://user-images.githubusercontent.com/60404243/222794246-583bfd59-1d9f-4e59-a99c-cfc70450a26b.png)

**Reply by melohagan** [reply]
Hey @AlexandriaLittle 

There may be a bug with automation looping at the moment: https://github.com/Budibase/budibase/discussions/9821
